packages/discord.js/src/client/BaseClient.js

```diff
@@ -18,6 +18,12 @@ class BaseClient extends EventEmitter {
       throw new DiscordjsTypeError(ErrorCodes.InvalidType, 'options', 'object', true);
     }
 
+    if (options.rest?.userAgentAppendix !== undefined) {
```

Comment:
Appending an empty string is probably not very useful
```suggestion
    if (options.rest?.userAgentAppendix) {
```